insights/combiners/tmpfilesd.py

```diff
@@ -0,0 +1,74 @@
+"""
+Tmpfilesd configuration
+=======================
+
+The tmpfilesd files are normally available to rules as a list of TmpFilesd
+objects. This combiner turns those into one set of data, and provides a
+``find_file()`` method to search for that filename among all the files.
+"""
+
+from insights.core.plugins import combiner
+from insights.parsers.tmpfilesd import TmpFilesD
+
+
+@combiner(TmpFilesD)
+class AllTmpFiles(object):
+    """
+    Combiner for accessing all the tmpfilesd configuration files. Configuration
+    files can be found in three directories: ``/usr/lib/tmpfiles.d``,
+    ``/run/tmpfiles.d``, and ``/etc/tmpfiles.d``. Each directory overrides the
+    settings in the previous directory. This combiner checks for and accounts
+    for this behavior.
+
+    Attributes:
+        files(set): the set of files found in all data files.
+        rules(dict): a dictionary of rules using the config file as the key
+        file_paths(list): a list of the file paths for the configurations.
+    """
+    def __init__(self, tmpfiles):
+        rules = {}
+        files = set([])
+        self.tmpfiles = tmpfiles
+        file_paths = []
+        # Files must be sorted so that /etc/tmpfiles.d overrides /run/tmpfiles.d
+        # which overides /usr/lib/tmpfiles.d. To do this, we have to list file
+        # paths in reverse order.
+        for tmpfile in sorted(tmpfiles, key=lambda f: f.file_path, reverse=True):
+            file_paths.append(tmpfile.file_path)
+            rules[tmpfile.file_path] = []
+            for new_rule in tmpfile.rules:
+                _file = (new_rule['path'])
+                if _file in files:
+                    [rules[tmpfile.file_path].remove(_rule) for _rule in rules[tmpfile.file_path] if _rule['path'] == _file]
+                rules[tmpfile.file_path].append(new_rule)
+            files.update(tmpfile.files)
+
+        self.rules = rules
+        self.files = files
+        self.file_paths = file_paths
+        super(AllTmpFiles, self).__init__()
+
+    def find_file(self, filename):
+        """
+        Find all the rules matching a given file. Using the original match
+        method from the parser.
+
+        Examples:
+            >>> data = tmpfiles
+            >>> results = data.find_file('/tmp/sap.conf')
+            >>> len(results)
+            1
+            >>> results[0]['path']
+            '/tmp/sap.conf'
+            >>> results[0]
+            {'path': '/tmp/sap.conf', 'mode': '644', 'type': 'x', 'age': None,
+             'gid': None, 'uid': None, 'argument': None}
+
+        Parameters:
+            filename(str): File path for the file to be searched in the rules
+
+        Returns:
+            (list): a list of rules where the path matches the file searched
+            for.
+        """
+        return [match for data in self.tmpfiles for match in data.find_file(filename)]
```

Comment:
Do you intend to return a valid `rule` via this method?
As per current code, it does not return the valid rule (when there are duplicates) but returns all of them. The plugin developers will have to do a further check to get the valid one. (And the `file_paths` is not returned together, it seems difficult to do the check)

Reply:
And better use the parsed data did by this `combiner` to implement this method.